src/test/groovy/org/mockito/release/gradle/ReleaseNeededTaskTest.groovy

```diff
@@ -4,12 +4,47 @@ import org.gradle.api.GradleException
 import org.gradle.testfixtures.ProjectBuilder
 import org.mockito.release.internal.comparison.PublicationsComparatorTask
 import spock.lang.Specification
+import spock.lang.Unroll
 
 class ReleaseNeededTaskTest extends Specification {
 
     def tasks = new ProjectBuilder().build().getTasks();
     ReleaseNeededTask underTest = tasks.create("releaseNeeded", ReleaseNeededTask)
 
+    @Unroll
+    def "release is needed" (commitMessage, branch, pullRequest, publicationsComparatorsResults, releaseNotNeeded){
+        given:
+        underTest.setCommitMessage(commitMessage)
+        underTest.setBranch(branch)
+        underTest.setPullRequest(pullRequest)
+        int i = 0;
+        publicationsComparatorsResults.each{
+            def task = tasks.create("" + i++, PublicationsComparatorTask)
+            task.setPublicationsEqual(it)
+            underTest.addPublicationsComparator(task)
+        }
+        underTest.setReleasableBranchRegex("master")
+
+        expect:
+        underTest.releaseNeeded()
+        
+        underTest.isReleaseNotNeeded() == releaseNotNeeded
```

Comment:
Nitpick: can we get rid of "isReleaseNotNeeded()" method and instead just make "releaseNeeded()" return boolean? This will simplify the code a little bit.